### PR TITLE
Add uSMART Trade v3.20.0

### PR DESCRIPTION
**Pull request by chunheisiu**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Comment by razvanazamfirei** [comment]
Hi @chunheisiu, thank you for submitting the PR. It looks that the homepage has multiple country-specific versions, including https://www.usmartglobal.com/. Are there the apps also different?

**Comment by chunheisiu** [comment]
> Hi @chunheisiu, thank you for submitting the PR. It looks that the homepage has multiple country-specific versions, including https://www.usmartglobal.com/. Are there the apps also different?

Both the download links on the global site and the Hong Kong site point to the same URL, so I believe it is safe to assume they are identical.

File: Casks/usmart-trade.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+cask "usmart-trade" do
+  version "3.20.0,1a879021-a3eb-4059-b141-5760ac163737,2023-03-24"
+  sha256 "a48955eb6df082ccc6e1612cf213c7dd60c50ece4d2f1eeff8f6cc06bcf02888"
+
+  url "https://jy-common-prd-1257884527.cos.ap-guangzhou.myqcloud.com/admin/app-version-file/#{version.csv.third}/#{version.csv.second}/uSMART%20Trade-#{version.csv.first}.dmg",
```
Comment (razvanazamfirei) [inline Casks/usmart-trade.rb:5]: This is another obfuscated URL. See([#143196](https://github.com/Homebrew/homebrew-cask/pull/143196)[#143572](https://github.com/Homebrew/homebrew-cask/pull/143572))
Their API (https://jy.yxzq.com/config-manager-admin/api/get-latest-package/v1) serves the latest version and download link.

```shell
curl 'https://jy.yxzq.com/config-manager-admin/api/get-latest-package/v1' \
        -H 'content-type: application/json;charset=UTF-8' \
        --data-raw '[{"appType":7,"system":"mac"}]'
```
```json
{
    "code": 0,
    "msg": "调用成功",
    "data": [
        {
            "appType": "7",
            "system": "mac",
            "url": "https://jy-common-prd-1257884527.cos.ap-guangzhou.myqcloud.com/admin/app-version-file/2023-03-24/1a879021-a3eb-4059-b141-5760ac163737/uSMART Trade-3.20.0.dmg"
        }
    ]
}
```

I haven't been able to use the response in the url and I'm afraid the current url will not be stable.

File: Casks/usmart-trade.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      verified: "jy-common-prd-1257884527.cos.ap-guangzhou.myqcloud.com/admin/app-version-file/"
```
Comment (razvanazamfirei) [inline Casks/usmart-trade.rb:6]:
```suggestion
      verified: "jy-common-prd-1257884527.cos.ap-guangzhou.myqcloud.com/"
```

File: Casks/usmart-trade.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  name "uSMART Trade"
+  desc "Trade Stocks & Options"
+  homepage "https://www.usmartglobal.com/"
+
+  livecheck do
+    url "https://www.usmart.hk/en/download"
+    regex(%r{
```
Comment (razvanazamfirei) [inline Casks/usmart-trade.rb:13]:
```suggestion
    regex(%r{href=['"].*?file/(\d+[-_]\d+[-_]\d+)/((?:\w+-)+(?:\w+))/.*?Trade[-_]?(\d+(?:\.\d+)+).dmg}i)
```

File: Casks/usmart-trade.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      /(\d{4}\b[._-]\d{2}\b[._-]\d{2}\b)
+      /([0-9a-f]{8}\b-[0-9a-f]{4}\b-[0-9a-f]{4}\b-[0-9a-f]{4}\b-[0-9a-f]{12})
+      /uSMART\sTrade[._-](\d+(?:\.\d+)+)\.dmg
+    }ix)
```
Comment (razvanazamfirei) [inline Casks/usmart-trade.rb:17]:
```suggestion
    regex(%r{href=['"].*?file/(\d+[-_]\d+[-_]\d+)/((?:\w+-)+(?:\w+))/.*?Trade[-_]?(\d+(?:\.\d+)+).dmg}i)
```

File: Casks/usmart-trade.rb (hunk 1 above)
Comment (chunheisiu) [reply Casks/usmart-trade.rb:5]: I see. Do you suggest using this URL for livecheck?

Comment (razvanazamfirei) [reply Casks/usmart-trade.rb:5]: I think this is fine so far (the API will just add complexity), but in case it breaks we can look into it more.

**Comment by razvanazamfirei** [comment]
\rebase

**Review by razvanazamfirei — DISMISSED** [review]

**Review by razvanazamfirei — APPROVED** [review]

**Comment by razvanazamfirei** [comment]
@chunheisiu, thank you so much for your contribution and patience while we worked through this!

**Pull request merged**